### PR TITLE
Fixes #461: Implement own search city dialog using bottom sheet

### LINKED ISSUE #461: Implement own search city dialog using bottom sheet
### Current implementaion
We are using this library: https://github.com/mirrajabi/search-dialog

### What to do?
Remove the use of the above library, and implement our own search city dialog in a bottom sheet

### PR DESCRIPTION
**Pull request by msecilmis**
## Description

Removed the search library used for city filtering, instead implemented own filter and bottom sheet dialog fragment.

Fixes #461 Implement own search city dialog using bottom sheet

Screenshot_20190903-203606 1
Screenshot_20190903-203622 1
Screenshot_20190903-203639 1
Screenshot_20190903-203644 1
Screenshot_20190903-203650 1

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ x] `./gradlew assembleDebug assembleRelease`
- [x ] `./gradlew checkstyle`

# Checklist:
- [ x] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
@msecilmis Please add screenshots too

**Comment by msecilmis** [comment]
updated the pull request

File: Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+            holder.tvCityName.setText(sb);
+            Picasso.with(context).load(citySearchModel.getImageUrl())
+                    .into(holder.civCityImage);
+
+            if (mListener != null)
+                holder.itemView.setOnClickListener(
+                        view -> mListener.onCitySearchModelClicked(position));
+
+
+//            holder.text.setText(String.valueOf(position));
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java]: remove this code

File: Android/app/src/main/res/layout/fragment_citysearchmodel_list_dialog.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            android:textStyle="bold" />
+
+        <EditText
+            android:id="@+id/et_query"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:drawableEnd="@drawable/search_gray"
+            android:layout_marginLeft="12dp"
+            android:layout_marginRight="12dp"
+            tools:hint="What are you looking for?" />
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/fragment_citysearchmodel_list_dialog.xml]: don't add constant strings here directly. Add these to strings.xml

File: Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java (hunk 1 above)
Comment (msecilmis) [reply Android/app/src/main/java/io/github/project_travel_mate/searchcitydialog/CitySearchBottomSheetDialogFragment.java]: removed

File: Android/app/src/main/res/layout/fragment_citysearchmodel_list_dialog.xml (hunk 2 above)
Comment (msecilmis) [reply Android/app/src/main/res/layout/fragment_citysearchmodel_list_dialog.xml]: used string resource instead of constant string

**Comment by msecilmis** [comment]
made requested changes

**Pull request merged**